backend/api/Services/RobotService.cs

```diff
@@ -98,29 +99,142 @@ public async Task<Robot> CreateFromQuery(CreateRobotQuery robotQuery)
             throw new DbUpdateException("Could not create new robot in database as robot model does not exist");
         }
 
-        public async Task<Robot> UpdateRobotStatus(string robotId, RobotStatus status)
+        public async Task<Robot?> UpdateRobotStatus(string robotId, RobotStatus status)
         {
-            logger.LogInformation("Updating robot {robotId} status to {status}", robotId, status);
-            return await UpdateRobotProperty(robotId, "Status", status);
+            var robotQuery = context.Robots.Where(robot => robot.Id == robotId).Include(robot => robot.CurrentInstallation);
+            var robot = await robotQuery.FirstOrDefaultAsync();
+
+            await VerifyThatUserIsAuthorizedToUpdateDataForInstallation(robot!.CurrentInstallation);
+
+            await robotQuery.ExecuteUpdateAsync(robots => robots.SetProperty(r => r.Status, status));
+
+            robot = await robotQuery.FirstOrDefaultAsync();
+            NotifySignalROfUpdatedRobot(robot!, robot!.CurrentInstallation!);
```

Comment:
Why are we stating in the function return value that the returned Robot can be null, whilst here we are stating that it won't be null?

Reply:
This function was missing the throw if null function calls. Fixed now.